### PR TITLE
storage/driver: decrease memory allocation done during testsuite

### PR DESCRIPTION
**Pull request by stevvooe**
Signed-off-by: Stephen J Day <stephen.day@docker.com>

### CODE REVIEW

**Comment by stevvooe** [comment]
cc @aaronlehmann

**Comment by stevvooe** [comment]
I don't understand how this is taking longer but it is. I get nearly 25% speed up on short tests locally. It must be the remote fs targets are taking more time. We'll have to see if is actually taking longer for things like filesystem test or it is just service availability.

File: registry/storage/driver/testsuites/testsuites.go [hunk 1]
```diff
@@ -1135,12 +1135,19 @@ func randomFilename(length int64) string {
...
-	for i := range b {
-		b[i] = byte(rand.Intn(2 << 8))
+// randomBytes pre-allocates all of the memory sizes needed for the test. If
+// anything panics while accessing randomBytes, just make this number bigger.
+var randomBytes = make([]byte, 96<<20)
+
+func init() {
+	// increase the random bytes to the required maximum
+	for i := range randomBytes {
+		randomBytes[i] = byte(rand.Intn(2 << 8))
```
Comment (aaronlehmann) [inline registry/storage/driver/testsuites/testsuites.go]: It's likely faster to generate 32-bit or 64-bit random values and split those across multiple bytes. I imagine this initialization is not much of a bottleneck, though, so this seems fine.

File: registry/storage/driver/testsuites/testsuites.go [hunk 2]
```diff
@@ -1135,12 +1135,19 @@ func randomFilename(length int64) string {
...
 	}
-	return b
+}
+
+func randomContents(length int64) []byte {
```
Comment (aaronlehmann) [inline registry/storage/driver/testsuites/testsuites.go]: It seems weird for this to take an `int64` if it's using the argument to slice an array. Changing the type to `int` would avoid a cast.

File: registry/storage/driver/testsuites/testsuites.go (hunk 1 above)
Comment (stevvooe) [inline registry/storage/driver/testsuites/testsuites.go]: The bottleneck is the allocation and garbage collection, as far as I can tell.

File: registry/storage/driver/testsuites/testsuites.go (hunk 2 above)
Comment (stevvooe) [inline registry/storage/driver/testsuites/testsuites.go]: The rest of the test code calls this directly with an int64. I'm actually surprised this compiles.

**Comment by stevvooe** [comment]
Kicking off the tests again to see if we just hit a slow set of driver tests. If we don't see this below 18m, I'm not sure if it is an improvement. The memory usage does go way down, so that may be enough to push this through.

**Comment by stevvooe** [comment]
@aaronlehmann @RichardScothern Merging this is up to you. There seems to be a slight improvement, but it is likely we are IO limited on testing targets. The next step to reducing runtime will be reducing the amount of data we send to backends for testing.

**Comment by RichardScothern** [comment]
If it's a performance boost locally and reduces memory then it's likely to help future tuning.   LGTM.

**Pull request merged**